### PR TITLE
WIP JW-885 ksync

### PR DESCRIPTION
**Pull request by m-kostrzewa**

### CODE REVIEW

File: src/ksync/ksync_sock.cc [hunk 1]
```diff
@@ -627,10 +649,38 @@ KSyncSockNetlink::KSyncSockNetlink(boost::asio::io_service &ios, int protocol)
...
 
 KSyncSockNetlink::~KSyncSockNetlink() {
+#ifdef _WINDOWS
+    boost::system::error_code ec;
+    pipe_.close(ec);
+    if (ec) {
+        LOG(ERROR, "Error closing KSync pipe: " << ec);
+        assert(0);
+    }
+#endif
```
Comment (m-kostrzewa) [inline src/ksync/ksync_sock.cc:658]: See where is the Linux destructor code

**Comment by m-kostrzewa** [comment]
I've opened a PR to this branch with Transmit Queue related changes: https://github.com/codilime/contrail-controller/pull/28

**Comment by m-kostrzewa** [comment]
Janie, przetestuj proszę

File: src/vnsw/agent/test/test_init.cc [hunk 2]
```diff
@@ -17,26 +17,26 @@
...
-    pthread_attr_t attr;
     int ret;
-
-    pthread_attr_init(&attr);
-    if ((ret = pthread_create(&asio_thread, &attr, asio_poll, NULL)) != 0) {
+    try {
+        asio_thread = boost::thread(asio_poll);
+    }
+    catch (const std::exception &e) {
         LOG(ERROR, "pthread_create error : " <<  strerror(ret) );
```
Comment (jacek-izykowski) [inline src/vnsw/agent/test/test_init.cc:33]: It is not "pthread_create error" anymore.

Comment (jacek-izykowski) [inline src/vnsw/agent/test/test_init.cc:33]: "ret" is not set properly.

**Comment by sodar** [comment]
This PR was divided into 2 independent PRs:

* ksync tests: https://github.com/codilime/contrail-controller/pull/33
* ksync porting: https://github.com/codilime/contrail-controller/pull/34

**Pull request closed without merging**